### ISSUE TITLE
Duplicate identifier 'Promise'.

### ISSUE THREAD

**Issue by Waog** [palantir/eclipse-typescript#257]
Problem
--
I get the following compile error in ``es6-promise.d.ts`` from [DefinitelyTyped](https://github.com/borisyankov/DefinitelyTyped/blob/master/es6-promise/es6-promise.d.ts):
*Duplicate identifier 'Promise'.*

Those two lines are marked with the error:
`declare class Promise<R> implements Thenable<R> {`
`declare module Promise {`

This error only occurs in eclipse with the eclipse-typescript plugin. When using `tsc` from the command line no errors are reported.

Project typescript plugin settings
--
```
build.path.exportedFolder=
build.path.sourceFolder=app/ts;typings;test/spec
compiler.codeGenTarget=ECMASCRIPT5
compiler.compileOnSave=false
compiler.generateDeclarationFiles=false
compiler.mapSourceFiles=false
compiler.moduleGenTarget=COMMONSJS
compiler.noEmitOnError=true
compiler.noImplicitAny=false
compiler.noLib=false
compiler.outputDirOption=
compiler.outputFileOption=
compiler.removeComments=false
compiler.suppressImplicitAnyIndexErrors=false
eclipse.preferences.version=1
```

tsc compiler settings
--
I'm using a grunt task:
```js
options: {
  target: 'es5',
  sourceMap: true,
  declaration: true,
  module: 'commonjs',
  emitDecoratorMetadata: true
}
```

**Comment by derekcicerone** [comment]
Thanks for the report! I think this a duplicate of #241. I'm working on a fix now.

**Comment by Waog** [comment]
So it should work with the newest version?

**Comment by derekcicerone** [comment]
Yup!

**Comment by Waog** [comment]
Hmm... either it is fixed or hidden behind the new errors occurring. I'll open two followup tickets...

**Comment by derekcicerone** [comment]
Are you on the main releases? I just updated to TS 1.5 so new errors may be due to better error-finding logic in TS itself.

**Comment by Waog** [comment]
I'm still on the dev release and updated using the eclipse plugin update mecanism. And it seems to be one of the newer errors, yes.

**Comment by derekcicerone** [comment]
Ah yeah, the fix isn't available in the develop release. The main release is now the best one - I'd suggest switching. You can change the update site to remove "develop" and then just look for updates.

**Comment by Waog** [comment]
No updates where found when switching like you suggested, but uninstalling and installing the plugin worked.

**Comment by derekcicerone** [comment]
Oh weird. Ok cool.
